src/JuliusSweetland.OptiKey.Core/UI/ViewModels/MainViewModel.ServiceEventHandlers.cs

```diff
@@ -2372,47 +2380,117 @@ private void SelectVoice(string voice)
             NavigateToMenu();
         }
 
-        private void StepList(string command)
+        public void StepList(KeyValue singleKeyValue, List<string> multiKeySelection)
+        {
+            if (!string.IsNullOrEmpty(singleKeyValue.String))
+            {
+                Log.InfoFormat("StepList called with singleKeyValue [{0}]", singleKeyValue.String);
+
+                List<string> commandList = singleKeyValue.String.Split('<').ToList();
+                if (commandList.Any())
+                {
+                    var cTask = Task.Run(() => StepListCommands(singleKeyValue, multiKeySelection, commandList));
+                }
+            }
+        }
+
+        private async Task<int> StepListCommands(KeyValue singleKeyValue, List<string> multiKeySelection, List<string> commandList)
         {
-            if (!string.IsNullOrEmpty(command))
+            Log.InfoFormat("StepListCommands called with a list of {0} commands", commandList.Count);
+            if (commandList.Contains("Wait>"))
             {
-                KeyValue singleKeyValue;
-                string[] stringSeparators = new string[] { "<" };
-                foreach (var vStep in command.Split(stringSeparators, StringSplitOptions.None).ToList())
+                //if there is a wait command then make this task async by using await
+                await Task.Delay(10);
```

Comment:
@JuliusSweetland I only break into async if there is a "Wait" command associated with the key. Worth noting, regardless of this being sync or async, it is always on it's own thread and I haven't traced things down yet, but it seems there is interference with callbacks.